components/collector/src/source_collectors/dependency_track/security_warnings.py

```diff
@@ -44,8 +45,11 @@ async def _get_source_responses(self, *urls: URL) -> SourceResponses:
         """Extend to get the findings."""
         api_url = await self._api_url()
         self._projects = await self._get_projects_by_uuid()
-        project_finding_urls = [URL(f"{api_url}/finding/project/{uuid}") for uuid in self._projects]
-        return await super()._get_source_responses(*project_finding_urls)
+        if self._projects:
+            project_finding_urls = [URL(f"{api_url}/finding/project/{uuid}") for uuid in self._projects]
+            return await super()._get_source_responses(*project_finding_urls)
+        error_message = "No projects found"
```

Comment:
This is duplicated code, can it not fit in shared parent `DependencyTrackLatestVersionStatusBase` or even higher?

Reply:
Yeah, you're right. Moved to `DependencyTrackBase`.